src/framework/preferences/js/PrefsEditor.js

```diff
@@ -391,13 +391,17 @@ var fluid_2_0 = fluid_2_0 || {};
             fluid.model.diff(userSelections.preferences, fluid.get(that.initialModel, ["preferences"]), stats);
 
             fluid.each(stats.changeMap, function (state, path) {
-                fluid.set(changedSelections, ["preferences", path], userSelections.preferences[path]);
+                fluid.set(changedPrefs, path, userSelections.preferences[path]);
```

Comment:
I refactored the save function to save for specific model path. As long as the path is relative to the model root, we are able to find and handle preferences path specially. Let me know if the refactored is what you were thinking. Thanks.